compose/compose-file/index.md

```diff
@@ -691,6 +699,23 @@ If you have further questions, please refer to the discussion on the GitHub
 issue [docker/compose/4513](https://github.com/docker/compose/issues/4513){: target="_blank" class="_"}.
 {: .important}
 
+##### Generic Resources
+
+Generic resources are a way to select the kind of nodes your task can land on.
+
+In a swarm cluster, nodes can advertise Generic resources as Discrete values or
+as Named values such as SSD=3 or GPU=UID1, GPU=UID2, ...
+
+The Generic resources on a service allows you to request for a number of these
+Generic resources advertised by swarm nodes and have your tasks land on nodes
+with enough available resources to statisfy your request.
+
+If you requested Named Generic resource(s), the resources selected will be
+exposed in your container through the use of environment variables.
+E.g: `DOCKER_RESOURCE_GPU=UID1,UID2`
+
+You can only set the `generic_resources` field for reservations.
```

Comment:
Is there a place to link to for what "reservations" are? Or would you like to add it to the glossary in `_data/glossary.yml`?

Reply:
Well there is the resource example just in the previous section.

I've updated the sentence to make it more explicit that it's a resource field.